Migrate to a different (new) skeleton

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request migrates this repository's upstream skeleton from [cisagov/skeleton-aws-lambda] to [cisagov/skeleton-aws-lambda-python]. This closes #50.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

The [cisagov/skeleton-aws-lambda] project has been archived in favor of the new [cisagov/skeleton-aws-lambda-python] skeleton. This skeleton better aligns with new options for building lambdas from AWS and reorganizes Python lambda projects to be about building deployment artifacts vs housing Python packages that also get built into deployment artifacts. As a result downstream repositories like this one must be updated.

### ⚠ Note ###

This is essentially a lift-and-shift migration. This project still needs a rework once this pull request is merged to better align the existing functionality with the new skeleton to truly be "finished".
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I verified that the generated build artifact correctly processed data in my testing environment.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Add a tag or create a release.

[cisagov/skeleton-aws-lambda]: https://github.com/cisagov/skeleton-aws-lambda
[cisagov/skeleton-aws-lambda-python]: https://github.com/cisagov/skeleton-aws-lambda-python

dav3r committed:
Enable terraform validate pre-commit hook

This hook should now work as expected in most, if not all of our repos 
now that we are finally updating to terraform 0.13.x (on our way to 
1.0.x).

dav3r committed:
Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when testing is completed.

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Revert "Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit 6a7fbf07bd371d0493c523ce24647e5c04c77c03.

dav3r committed:
Merge pull request #90 from cisagov/improvement/enable_terraform_validate

Enable terraform validate pre-commit hook

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Use an id when using cisagov/setup-env-github-action

mcdonnnj committed:
Use setup-env outputs for Terraform version

Change from using an environment variable to the outputs of the
cisagov/setup-env-github-action instead.

mcdonnnj committed:
Use setup-env outputs for Packer version

We use a `PACKER_VERSION` environment variable for the "Install Packer" step
that is populated from the `setup-env` outputs to get around `yamllint` lint
length limits.

mcdonnnj committed:
Use setup-env outputs for shfmt version

We change the "Install shfmt" step to use two local environment variables to
provide the package's URL and version to install. This allows us to work around
`yamllint` line length limits.

mcdonnnj committed:
Use setup-env outputs for Terraform-docs version

We change the "Install Terraform-docs" step to use two local environment
variables to provide the package's URL and version to install. This allows us
to work around `yamllint` line length limits.

mcdonnnj committed:
Store Go version as a step output

This changes from using an environment variable to using a step output to store
the Go version that is installed. This mirrors changes made to the other
program versions and how they're stored.

mcdonnnj committed:
Update the ansible-lint version for pre-commit

mcdonnnj committed:
Add an ansible-lint configuration file

This file is copied from cisagov/skeleton-ansible-role#85 and tweaked to pass
pre-commit. See here for for a list of the elements that can exist in this
file: https://ansible-lint.readthedocs.io/en/latest/configuring.html

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add dependabot configuration for Terraform

This will configure `dependabot` to scan Terraform configurations if they exist
in a repository. If a repository stores a Terraform configuration in a sub-
directory this configuration will need to be modified or an additional
configuration must be added if there still exists a Terraform configuration in
the root directory.

mcdonnnj committed:
Merge pull request #91 from cisagov/improvement/pre-commit_update

Update `pre-commit` Hooks

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/update_ansible-lint_pre-commit_version

Update the `ansible-lint` Version in the pre-commit Configuration

mcdonnnj committed:
Merge pull request #94 from cisagov/improvement/add_dependabot_configuration_for_terraform

Add `dependabot` Configuration for Terraform

mcdonnnj committed:
Merge pull request #92 from cisagov/improvement/use_action_outputs

Use Outputs of cisagov/setup-env-github-action to Control Installed Program Versions

mcdonnnj committed:
Update ansible-lint configuration

Remove the manual exclusion of the `converge.yml` file in the default molecule
scenario. With the changes in cisagov/skeleton-ansible-role#88 this exclusion
is no longer necessary.

mcdonnnj committed:
Update ansible-lint configuration

Instead of excluding molecule configurations for systemd individually we
instead direct ansible-lint to process any of these configurations in
scenario directories as plain yaml files. This mirrors how ansible-lint
handles `molecule.yml` files in scenario directories by default.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Merge pull request #97 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/update_ansible-lint_configuration

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_ansible-lint_configuration

Update `ansible-lint` Configuration

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Add comment to Dependabot configuration

This comment explains that the configuration may have commented out
ignore directives that should be uncommented in downstream projects.

mcdonnnj committed:
Disable comments-indentation rule for yamllint

yamllint does not like it when you comment out pieces of dictionaries
in lists. Upcoming additions to the Dependabot configuration will run
afoul of this so we are updating the yamllint configuration.

mcdonnnj committed:
Merge pull request #99 from cisagov/improvement/update_dependabot_and_yamllint_configurations

Add boilerplate for upcoming Dependabot configuration changes

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #98 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

mcdonnnj committed:
Use asterisks for emphasis- and strong-styles

Given the inconsistent behavior of underscores used within words for
style we should prefer asterisks now that these rules are available.

mcdonnnj committed:
Add missing hyphen in markdownlint rule header

mcdonnnj committed:
Merge pull request #100 from cisagov/improvement/update_markdownlint_configuration

Use new `markdownlint` rules for emphasis- and strong-styles

dependabot[bot] committed:
Bump actions/setup-python from 2 to 3

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Add Dependabot ignore directives

This adds commented out ignore directives for the following GitHub
Actions:

- action/cache
- action/checkout
- action/setup-python

These should be uncommented downstream to ensure that updates to these
dependencies are pushed from pull requests made in the skeleton.

mcdonnnj committed:
Upgrade from Python 3.9 to 3.10 for the `lint` job in GHA

mcdonnnj committed:
Use consistent quoting for software versions

mcdonnnj committed:
Merge pull request #101 from cisagov/dependabot/github_actions/actions/setup-python-3

Bump actions/setup-python from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/checkout-3

mcdonnnj committed:
Merge pull request #102 from cisagov/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

dependabot[bot] committed:
Bump actions/cache from 2 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge branch 'develop' into improvement/add_dependabot_ignores_for_github_actions

mcdonnnj committed:
Merge pull request #107 from cisagov/improvement/add_dependabot_ignores_for_github_actions

Add Dependabot ignore directives

mcdonnnj committed:
Merge pull request #104 from cisagov/dependabot/github_actions/actions/cache-3

Bump actions/cache from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into improvement/use_python_3.10_for_gha

mcdonnnj committed:
Merge pull request #106 from cisagov/improvement/use_python_3.10_for_gha

Use Python 3.10 for the `lint` job in GitHub Actions

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #105 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Add pre-commit hooks to validate GitHub Actions content

Add hooks to validate GitHub Actions and GitHub Action workflows. These
are validated using JSON Schemas.

mcdonnnj committed:
Merge pull request #109 from cisagov/improvement/add_github_actions_pre-commit_hooks

Add `pre-commit` hooks to validate GitHub Actions content

mcdonnnj committed:
Rename repository references after clone

mcdonnnj committed:
Add lineage configuration

mcdonnnj committed:
Add Docker scaffolding

Add the Dockerfile and appropriate Docker Compose configuration for
generating a deployment package.

mcdonnnj committed:
Add the script to build a deployment package artifact

This script creates a deployment package zipfile inside the built
Docker image. The output directory should be mapped to the host
filesystem to retrieve the artifact. This is automatically handled
in the `docker-compose.yml` file.

mcdonnnj committed:
Add Pipfiles for each supported Python version

Since we are using Pipfiles (and Pipfile.lock files) to manage project
dependencies I have added pipenv to the development requirements in
`requirements-dev.txt`.

mcdonnnj committed:
Add Lambda handler

Implement an example handler for this project to show how functionality
could be handled.

mcdonnnj committed:
Update the Lambda build process

The build process is updated so that a functional Lambda image is
created as a base. The `docker-compose.yml` file was updated to provide
a service to create a deployment package for use in AWS and a service
to run the Lambda locally for testing.

mcdonnnj committed:
Add a `build` job to the GitHub Actions workflow

This job will create Lambda deployment packages for each supported
Python version and make them available as artifacts.

mcdonnnj committed:
Update project README

Update the README's introductory paragraph and add information about
building deployment packages and running the Lambda locally.

mcdonnnj committed:
Add version management

Add a version file and appropriate scripts to manage and tag versions
for this project.

mcdonnnj committed:
Add a barebones testing harness

Set up a barebones pytest configuration. It would be nice to test the
handler but given how dependencies are managed this might make local
testing a pain point.

mcdonnnj committed:
Add a CodeQL workflow

mcdonnnj committed:
Add a release workflow

This adds a workflow that is triggered on prerelease/release
publishing. Right now it is set up to upload artifacts as Action
artifacts due to the archival of the actions/upload-release-asset
Action.

mcdonnnj committed:
Fix vendor label in Dockerfile

The vendor name pulled from cisagov/skeleton-aws-lambda did not include
the update from cisagov/skeleton-docker.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Update to use long flag names

We prefer to use long flag names when available to improve the
readability of command calls. This fixes some short flag name usage
in the Dockerfile and build_artifact.sh script that was missed.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Improve error messages for the `divide` task

Using `%s - %s` was somewhat ambiguous so a more verbose error message
will now be used.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Fix language in `lambda_handler.py`

Fix a missing period in a function comment and a typo in an error
message.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Update the bandit pre-commit hook entries

Add an appropriate comment for the hook that runs against the `tests/`
directory and correct the grammar of the comment for the hook that runs
against everything but the `tests/` directory. Update the names for the
two hook entries to mirror the comment changes.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Update `pytest.ini` to use long flag names

We prefer long flag names whenever possible and this adds a comment to
explain the flags used to cover the one short flag we must use.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add an ownership comment to the Dependabot configuration

Add a comment that states that the commented out ignore directives are
managed by cisagov/skeleton-generic.

mcdonnnj committed:
Add additional Dependabot ignore directives

This adds the other versioned Actions that should be managed by
cisagov/skeleton-generic to the list of commented out dependencies to
ignore.

mcdonnnj committed:
Update actions/setup-go from v2 to v3

Additionally as of v3.1.0 of actions/setup-go there is a go-version
output value to retrieve the version of Go installed by the Action.
This allows us to remove the step to manually retrieve this information
from the Go executable.

mcdonnnj committed:
Update the version of Go used from 1.16 to 1.18

Go 1.16 is no longer supported as of the release of 1.18 so it makes
sense to update to the latest version available.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Enable the allow-non-breakable-inline-mappings rule for yamllint

mcdonnnj committed:
Increase the maximum line length in the yamllint configuration

Increase the line length maximum from the default of 80 to 88. This 10%
increase mirrors the line length allowed in other linter configurations
such as black.

mcdonnnj committed:
Add a label configuration file

This configuration file stores information about the labels expected in
this repository.

mcdonnnj committed:
Add a workflow to synchronize repository labels

This adds a workflow to ensure that the repository labels are updated
to reflect changes to the label configuration file .github/labels.yml.

mcdonnnj committed:
Remove usage of set-output from our GitHub Actions workflow

GitHub has deprecated the set-output command per:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

This updates the GitHub Actions workflow to use the newly preferred
method to set the output for a job's step.

mcdonnnj committed:
Fix the repo of the flake8 pre-commit hook

Per PyCQA/flake8#1290 this hook moved from GitLab to GitHub. The
version we use is bumped to the latest tag on GitHub as well.

jsf9k committed:
Remove duplicate label

jsf9k committed:
Add a label for issues or PRs that involve test code

mcdonnnj committed:
Update the version of Go used from 1.18 to 1.19

Go 1.19 was released while this branch was in the wings and it makes
sense to bump to the latest Go release.

mcdonnnj committed:
Merge pull request #113 from cisagov/improvement/update_go_configuration

Update Go installation in the `build.yml` workflow

dependabot[bot] committed:
Bump hashicorp/setup-terraform from 1 to 2

Bumps [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) from 1 to 2.
- [Release notes](https://github.com/hashicorp/setup-terraform/releases)
- [Changelog](https://github.com/hashicorp/setup-terraform/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/setup-terraform/compare/v1...v2)

---
updated-dependencies:
- dependency-name: hashicorp/setup-terraform
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #111 from cisagov/dependabot/github_actions/hashicorp/setup-terraform-2

Bump hashicorp/setup-terraform from 1 to 2

dependabot[bot] committed:
Bump actions/setup-python from 3 to 4

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 3 to 4.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #115 from cisagov/dependabot/github_actions/actions/setup-python-4

Bump actions/setup-python from 3 to 4

mcdonnnj committed:
Merge pull request #112 from cisagov/improvement/update_dependabot_ignores

Update Dependabot ignore directives

mcdonnnj committed:
Merge pull request #119 from cisagov/bug/update_github_actions_workflow

Remove usage of `set-output` from our GitHub Actions workflow

mcdonnnj committed:
Merge pull request #117 from cisagov/improvement/update_yamllint_line-length_rules

Update the `line-length` configuration for `yamllint`

mcdonnnj committed:
Merge pull request #120 from cisagov/bug/verify_pre-commit_hook_repos

Update the `flake8` pre-commit hook configuration

mcdonnnj committed:
Merge pull request #114 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

mcdonnnj committed:
Merge pull request #118 from cisagov/improvement/add_label_sync_workflow

Add workflow to manage repository labels

mcdonnnj committed:
Fix GitHub label color values

Update the configuration for repository labels to remove the leading
`#` from color values. With a `#` leading the values they are seen as
invalid by the GitHub API.

mcdonnnj committed:
Correct spacing in yamllint rules

There was a missing empty line in the `.yamllint` file between two rule
definitions.

mcdonnnj committed:
Merge pull request #122 from cisagov/bug/fix_bad_configurations

Update two configuration files

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Enable dependabot ignore directives

Enable the dependabot ignore directives for GitHub Actions
dependencies.

mcdonnnj committed:
Add a `python` label to the repository

mcdonnnj committed:
Merge pull request #7 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Update Dockerfile label information

Pull in changes made to cisagov/skeleton-docker around the authors
label for consistency.

mcdonnnj committed:
Update the artifact production process

Fully divorce artifact creation from the Lambda image. We switch to
mounting the build_artifact.sh into the container as part of the
build_deployment_package service and move installing zip into that
script. We also remove the BUILD_FILE_NAME environment variable from
the Docker image and house it as part of the Docker Compose
configuration.

These changes better align the base image as a minimal AWS Lambda image
suited for use on its own. Building a deployment artifact now leverages
this image instead of being part of the image.

mcdonnnj committed:
Update dependency installation

Now that pipenv has the --extra-pip-args option we can pass the options
we need to the underlying pip calls. This allows us to use pipenv to
directly install the dependencies instead of the previous hack of using
a requirements.txt file generated from the Pipfile configuration.

mcdonnnj committed:
Add Python dependency maintenance instructions

Add a section about how to update Python dependencies for the Lambda to
the README.

mcdonnnj committed:
Update Python dependencies

Update the Python dependencies for the project by running `pipenv lock`
in each of the `src/py<Python version>` directories.

mcdonnnj committed:
Update missed GitHub Action versions

Update Action versions that were missed when Lineage updates were
merged.

mcdonnnj committed:
Add commented out dependabot ignore information

Add an attribution comment and a commented out ignore directive for
a `github-actions` dependency that should be managed by this project.
This ignore directive should be uncommented in downstream repositories.

mcdonnnj committed:
Remove the release.yml workflow

Work toward a fancy automated release process can be pushed off until
after the initial implementation is in place. This will allow
appropriate testing of an otherwise functional implementation so it is
appropriately isolated. In the interim releases can be managed by
manually creating a release and uploading the artifacts generated from
the merge commit's Actions run.

mcdonnnj committed:
Pull in updates to the bump_version.sh script

Pull in updates from cisagov/skeleton-packer around the regex used to
find the old version.

mcdonnnj committed:
Use a multi-stage build for the Lambda image

Part of this project's goals is to support the option of running a
bespoke image as a Lambda and not just using deployment packages. Since
we care about the created image due to this we should build it similar
to how we would build other Docker images. Thus we switch to a multi-
stage build so that the only things included in the final image are
what is absolutely necessary to run the Lambda (per the dependency
files). With this configuration we saw space savings of almost 10% by
switching to a multi-stage build.

mcdonnnj committed:
Ensure pipenv uses the system Python instead of a venv

The default behavior of pipenv is to create a virtual environment to
use for installing dependencies. Our configuration does not leverage
this venv since we are simply using pipenv to manage dependencies and
need all dependencies installed into the Lambda's task root directory.
This default behavior can create issues with our configuration because
if one of the core packages seeded into that venv (pip, setuptools, or
wheel) is the same version as a dependency in the lockfile then it will
not be installed into the Lambda's task root directory for later use..

mcdonnnj committed:
Enable changing the image tag used with Docker Compose

Add the ability to change the image tag used for the Lambda image in
the Docker Compose configuration. This is done using the LAMBDA_TAG
environment variable with a fallback to the default value of latest.

mcdonnnj committed:
Merge pull request #3 from cisagov/first-commits

Initial implementation

mcdonnnj committed:
Update lineage configuration

Update the lineage configuration to descend from the new
cisagov/skeleton-aws-lambda-python repository instead of the archived
cisagov/skeleton-aws-lambda repository.

mcdonnnj committed:
Merge github.com:cisagov/skeleton-aws-lambda-python into maintenance/migrate_to_new_skeleton

mcdonnnj committed:
Update the Lambda handler

Update the Lambda handler to incorporate the existing functionality
into the new template provided by cisagov/skeleton-aws-lambda-python.

mcdonnnj committed:
Dismantle the fdi package

Migrate the core functionality to the src/ directory with appropriate
updates. Delete the non-essential pieces of the Python package.

mcdonnnj committed:
Update the Pipfiles and generate new lockfiles

Update the respective Pipfiles to match the lambda's required packages.
Generate new lockfiles by running `pipenv lock` in each py3.*/
directory.

mcdonnnj committed:
Update the Docker configuration

Update the Docker configuration from the skeleton to match this
project.

mcdonnnj committed:
Update the isort configuration

The example package is no longer used so we do not need to list it as a
first party package.

mcdonnnj committed:
Update logging to be consistent

Update the logging in the core logic to match both our best practices
and the logging from the skeleton.

mcdonnnj committed:
Update the skip_record() helper function

Rework the function slightly to take in a format string as an argument
to better align with Python logging best practices.

mcdonnnj committed:
Enable dependabot ignore directive

Enable a dependabot ignore directive for GitHub Actions.

mcdonnnj committed:
Bump version from 1.0.0 to 1.1.0

mcdonnnj committed:
Bump version from 1.1.0 to 1.1.0-rc.1

mcdonnnj committed:
Remove the release workflow

This workflow does not exist in the new skeleton so we should remove it
as it is non-functional.

mcdonnnj commented:
The CodeQL tests are stuck because the workflow was disabled due to inactivity. They should kick off when I bump or finalize the version now that I've re-enabled the workflow.

dav3r commented:
> This is essentially a lift-and-shift migration. This project still needs a rework once this pull request is merged to better align the existing functionality with the new skeleton to truly be "finished".

Is there an issue (or issues) documenting what is needed here?

dav3r reviewed (approved):
Looks great- nice work! 👍

mcdonnnj commented:
> > This is essentially a lift-and-shift migration. This project still needs a rework once this pull request is merged to better align the existing functionality with the new skeleton to truly be "finished".
> 
> Is there an issue (or issues) documenting what is needed here?

I was waiting for any requested changes before creating the issue so I could be explicit in what I was detailing.

mcdonnnj committed:
Update the README

Ensure all non-root headers use sentence case. Update the information
about field mapping to reflect the changes from switching skeletons as
well as to improve readability.

jsf9k commented on README.md:
```diff
@@ -41,20 +27,74 @@ blank, the script will remove that JSON element from the record.
-is a very good idea to _always_ run the `docker-compose build` step when
-using this tool.
+The configuration in this repository allows you run the Lambda locally for
+testing as long as you do not need explicit permissions for other AWS
```
I think this word is slightly better here, but please disregard if you disagree.
```suggestion
testing as long as you do not require explicit permissions for other AWS
```

mcdonnnj replied:
This is pulled in from the skeleton so I will punt the change upstream.

jsf9k reviewed (approved):
I made one very minor suggestion, but this PR is approved even if you choose to disregard it.

dav3r reviewed (approved):
🎯

mcdonnnj committed:
Finalize version from 1.1.0-rc.1 to 1.1.0

mcdonnnj committed:
Remove unused version file

This file was added previously but was never used to track versioning
for this project.

mcdonnnj committed:
Remove coveralls configuration file

The new parent skeleton does not currently use coveralls so we should
remove this file for consistency.
